vars/cloudFoundryDeploy.groovy

```diff
@@ -196,11 +196,17 @@ private String deleteOptionIfRequired(Map config) {
     }
 }
 
-private String stopOldAppIfRequired(Map config) {
+private void stopOldAppIfRequired(Map config) {
+    String oldAppName = "${config.cloudFoundry.appName}-old"
+    String cfStopOutputFileName = "${UUID.randomUUID()}-cfStopOutput.txt"
+
     if (config.keepOldInstance && config.deployType == 'blue-green') {
-        return "cf stop ${config.cloudFoundry.appName}-old"
-    } else {
-        return ''
+        int cfStopReturncode = sh (returnStatus: true, script: "cf stop $oldAppName  &> $cfStopOutputFileName")
+        String cfStopOutput = readFile(file: cfStopOutputFileName)
```

Comment:
[minor] would be possible to move the read file after a check for return code != 0 since only in this case we really need the file, but it is agreed that it works also this way, hence the code can stay as it,  no request to change it, only a remark ... In this case also line 319 in the test could vanish ...

Reply:
Done.